Mathlib/LinearAlgebra/Projection.lean

```diff
@@ -424,6 +425,41 @@ theorem eq_conj_prod_map' {f : E →ₗ[R] E} (h : IsProj p f) :
   · simp only [coe_prodEquivOfIsCompl, comp_apply, coe_inr, coprod_apply, map_zero,
       coe_subtype, zero_add, map_coe_ker, prodMap_apply, zero_apply, add_zero]
 
+theorem isIdempotentElem {R V : Type*} [Semiring R] [AddCommMonoid V] [Module R V]
```

Comment:
This particular theorem is just the forward direction of `isProj_iff_isIdempotentElem`, so I might suggest moving it to directly after and giving it a proof like `T.isProj_iff_isIdempotentElem.mp ⟨U, h⟩` to emphasize the relation between the two.